app/topics/topic-management/add-existing-subtopic/add-existing-subtopic.component.ts

```diff
@@ -36,12 +36,26 @@ export class AddExistingSubtopicComponent {
     this.showChange.emit(this.show);
   }
 
+  private filterTopics(topics: any) {
+    console.log(topics)
+    for(let i = 0; i < this.subtopics.length; i++) {
```

Comment:
I think you can use `for (subtopic of this.subtopics)` syntax in the first loop. So it is easier to read.